communication/nats/dialog/dialog_establisher_test.go

```diff
@@ -20,50 +20,82 @@ func TestDialogEstablisher_Factory(t *testing.T) {
+
 	connection := nats.StartConnectionFake()
 	connection.MockResponse(
-		"provider1.dialog-create",
+		"peer-topic.dialog-create",
 		[]byte(`{
-			"payload": {
-				"reason": 200,
-				"reasonMessage": "OK"
-			},
-            "signature": "2Rg9KabJXdYEsMLynoeZ8+4cWjauHuZq/ydIE0NuNl1psu+AVz/8fHaqdG81CUgf2dNQHjciOVPagEb+X6//sgA="
-		}`))
+			"payload": {"reason":200,"reasonMessage":"OK"},
+            "signature": "iaV65n3kEve9+EzwWVi65qJFrb4FQZwq4yWdVH++abts3mW/xqKHpPKro7kX/liFRZgV5RHQMjE+TzPPdeJfewA="
+		}`),
+	)
 	defer connection.Close()
 
 	signer := &identity.SignerFake{}
+	establisher := mockEstablisher(myID, connection, signer)
 
-	establisher := &dialogEstablisher{
-		myIdentity: identity.FromAddress("consumer1"),
-		mySigner:   signer,
-		contactAddressFactory: func(contact dto_discovery.Contact) (*discovery.AddressNATS, error) {
-			assert.Exactly(t, dto_discovery.Contact{}, contact)
-			return discovery.NewAddressWithConnection(connection, "provider1"), nil
-		},
-	}
-	dialogInstance, err := establisher.CreateDialog(dto_discovery.Contact{})
+	dialogInstance, err := establisher.CreateDialog(peerID, dto_discovery.Contact{})
 	defer dialogInstance.Close()
 	assert.NoError(t, err)
 	assert.NotNil(t, dialogInstance)
 
-	dialogNats, ok := dialogInstance.(*dialog)
+	dialog, ok := dialogInstance.(*dialog)
 	assert.True(t, ok)
 
-	expectedCodec := NewCodecSecured(communication.NewCodecJSON(), signer, identity.NewVerifierSigned())
+	expectedCodec := NewCodecSecured(communication.NewCodecJSON(), signer, identity.NewVerifierIdentity(peerID))
 	assert.Equal(
 		t,
-		nats.NewSender(connection, expectedCodec, "provider1.consumer1"),
-		dialogNats.Sender,
+		nats.NewSender(connection, expectedCodec, "peer-topic."+myID.Address),
+		dialog.Sender,
 	)
 	assert.Equal(
 		t,
-		nats.NewReceiver(connection, expectedCodec, "provider1.consumer1"),
-		dialogNats.Receiver,
+		nats.NewReceiver(connection, expectedCodec, "peer-topic."+myID.Address),
+		dialog.Receiver,
+	)
+}
+
+func TestDialogEstablisher_CreateDialogWhenResponseHijacked(t *testing.T) {
+	myID := identity.FromAddress("0x6B21b441D0D2Fa1d86407977A3a5C6eD90Ff1A62")
+	peerID := identity.FromAddress("0x0d1a35e53b7f3478d00B7C23838C0D48b2a81017")
+
+	connection := nats.StartConnectionFake()
+	connection.MockResponse(
+		"peer-topic.dialog-create",
+		[]byte(`{
+			"payload": {"reason":200,"reasonMessage":"OK"},
+			"signature": "2Rg9KabJXdYEsMLynoeZ8+4cWjauHuZq/ydIE0NuNl1psu+AVz/8fHaqdG81CUgf2dNQHjciOVPagEb+X6//sgA="
+		}`),
 	)
+	defer connection.Close()
+
+	establisher := mockEstablisher(myID, connection, &identity.SignerFake{})
+
+	dialogInstance, err := establisher.CreateDialog(peerID, dto_discovery.Contact{})
+	defer dialogInstance.Close()
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "dialog creation error. failed to unpack response 'peer-topic.dialog-create'. invalid message signature ")
+	assert.Nil(t, dialogInstance)
+
+	_, ok := dialogInstance.(*dialog)
+	assert.True(t, ok)
```

Comment:
Since you already assert that `dialogInstance` is `nil` and this is error is present, what is the purpose of this type check?

Reply:
Fixed